src/ATen/native/xpu/SpectralOps.cpp

```diff
@@ -0,0 +1,40 @@
+#include <ATen/ATen.h>
+#include <ATen/native/xpu/mkl/SpectralOps.h>
+#include <ATen/xpu/XPUNativeFunctions.h>
+
+namespace at {
+
+Tensor XPUNativeFunctions::_fft_c2c(
+    const Tensor& self,
+    IntArrayRef dim,
+    int64_t normalization,
+    bool forward) {
+  TORCH_CHECK(self.is_complex());
+
+  if (dim.empty()) {
+    return self.clone();
+  }
+
+  return native::xpu::_fft_c2c_kernel(self, dim, normalization, forward);
+}
+
+Tensor& XPUNativeFunctions::_fft_c2c_out(
+    const Tensor& self,
+    IntArrayRef dim,
+    int64_t normalization,
+    bool forward,
+    Tensor& out) {
+  TORCH_CHECK(self.is_complex());
+
+  if (dim.empty()) {
+    out.copy_(self);
+    return out;
+  }
+
+  Tensor result =
+      native::xpu::_fft_c2c_kernel(self, dim, normalization, forward);
+  out.copy_(result);
```

Comment:
We have to alloc memory at native? Can we use user memory as output memory?

Reply:
Will discuss it with you offline.

Reply:
Just follow CUDA's design. Will try to refine this part of code in another PR.